src/measure_theory/borel_space.lean

```diff
@@ -1167,6 +1291,67 @@ lemma measurable_of_tendsto_metric {f : ℕ → α → β} {g : α → β}
   measurable g :=
 measurable_of_tendsto_metric' at_top hf lim at_top_countable_basis (λ i, countable_encodable _)
 
+lemma ae_measurable_of_tendsto_metric_ae {μ : measure α} {f : ℕ → α → β} {g : α → β}
+  (hf : ∀ n, ae_measurable (f n) μ)
+  (h_ae_tendsto : ∀ᵐ x ∂μ, filter.at_top.tendsto (λ n, f n x) (nhds (g x))) :
+  ae_measurable g μ :=
+begin
+  by_cases hβ : nonempty β,
+  { let p : α → (ℕ → β) → Prop := λ x f', filter.at_top.tendsto (λ n, f' n) (nhds (g x)),
+    let hp : ∀ᵐ x ∂μ, p x (λ n, f n x), from h_ae_tendsto,
+    let ae_seq_lim := λ x, ite (x ∈ ae_seq_set hf p) (g x) hβ.some,
+    refine ⟨ae_seq_lim, _, (ite_ae_eq_of_measure_compl_zero g (λ x, hβ.some)
+      (ae_seq_set hf p) (ae_seq.measure_compl_ae_seq_set_eq_zero hf hp)).symm⟩,
+    refine measurable_of_tendsto_metric (@ae_seq.measurable α β _ _ _ f μ hβ hf p) _,
+    refine tendsto_pi.mpr (λ x, _),
+    simp_rw [ae_seq, ae_seq_lim],
+    split_ifs with hx,
+    { simp_rw ae_seq.measurable_fun_eq_fun_of_mem_ae_seq_set hf hx,
+      exact @ae_seq.fun_prop_of_mem_ae_seq_set α β _ _ _ _ _ _ hβ hf x hx, },
+    { exact tendsto_const_nhds, }, },
+  { refine ⟨g, measurable_of_not_nonempty _ g, ae_eq_refl _⟩,
+    revert hβ,
+    exact mt (nonempty.map g), },
+end
+
+lemma measurable_of_tendsto_metric_ae {μ : measure α} [μ.is_complete] {f : ℕ → α → β} {g : α → β}
+  (hf : ∀ n, measurable (f n))
+  (h_ae_tendsto : ∀ᵐ x ∂μ, filter.at_top.tendsto (λ n, f n x) (nhds (g x))) :
+  measurable g :=
+ae_measurable_iff_measurable.mp
+  (ae_measurable_of_tendsto_metric_ae (λ i, (hf i).ae_measurable) h_ae_tendsto)
+
+lemma measurable_limit_of_tendsto_metric_ae [hβ : nonempty β] {μ : measure α} {f : ℕ → α → β}
```

Comment:
I don't think `nonempty β` is necessary here: if `α` is empty then there is nothing to do (take for `f_lim` the unique function defined on the empty space, and then the construction is true because there is nothing to check). Otherwise, `β` is also not empty because of the existence of `f`, and you can proceed with your proof.